_sub/monitoring/cloudwatch-alarms/log-anomaly/main.tf

```diff
@@ -0,0 +1,39 @@
+resource "aws_cloudwatch_metric_alarm" "log_anomaly" {
+  count = var.deploy ? 1 : 0
+
+  alarm_name          = "log-anomaly"
+  comparison_operator = "LessThanLowerThreshold"
+  evaluation_periods  = "5"
+  datapoints_to_alarm = "2"
+
+  metric_query {
+    id = "m1"
+    return_data = true
+
+    metric {
+      metric_name = "IncomingLogEvents"
+      namespace   = "AWS/Logs"
+      period      = var.check_period
+      stat        = "Sum"
+
+      dimensions = {
+        LogGroupName = "/k8s/hellman/kube-system"
+      }
+    }
+  }
+
+  metric_query {
+    id          = "ad1"
+    label       = "IncomingLogEvents (expected)"
+    expression  = "ANOMALY_DETECTION_BAND(m1, 1)"
```

Comment:
Is 1 standard deviation enough to trigger an alarm? Might be too sensitive.

Reply:
According to the graph, we never don't recieve logs unless there's a problem. But it's hard to find a sweet spot. Also, since we can't bind it to x amount of nodes, the deviation actually means more than 1 node are not sending log events.

Reply:
The standard deviation set at 1 essentially means that 1 out of 6 times the threshold might be crossed due to random variation. At 2, it should be down to 1 out of 40 times at 3 it should be at about 1 out of 200.

Since you are requiring two consecutive samples to fire the alarm the probability of the alarm firing due to random variation should be 1/6 * 1/6 = 1 out of 36. If the evaluation period is 5 minutes there are 288 periods within a day, so you would expect this to fire due to randomness ~8 a day.

If I am understanding correctly, it might be too sensitive, but give it a try I guess ...

![conspiracy](https://user-images.githubusercontent.com/2730310/233968290-662c5233-54dc-4e99-ac9f-3f0b09dd5987.jpg)

Reply:
The thing is that it measures the amount of logs and creates a buffer around that, so 1 node going down would not result in 1 deviation. Look here:
![image](https://user-images.githubusercontent.com/121795624/233978096-c2006051-1a19-423b-a2fd-a8294fd3283e.png)

Reply:
Okay looking further into it, it might be too sensitive.
![image](https://user-images.githubusercontent.com/121795624/233979323-4ddd66f7-acbd-4a02-bab9-4bbb9ad0481f.png)
This would have fired twice then